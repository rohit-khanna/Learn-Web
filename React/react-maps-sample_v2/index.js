import React from "react";
import { render } from "react-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import "./index.css";
import App from "./src/components/App";
import ConfigureStore from "./src/redux/configureStore";
import InitialState from "./src/redux/reducers/InitialState";
import { Provider as ReduxProvider } from "react-redux";

const store = new ConfigureStore(InitialState);

render(
  <ReduxProvider store={store}>
    <App />
  </ReduxProvider>,
  document.getElementById("app")
);
