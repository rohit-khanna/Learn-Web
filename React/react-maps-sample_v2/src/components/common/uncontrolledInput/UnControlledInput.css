.Input_ContainerNew {
  position: relative;
}

.Input_ContainerNew .icon {
  font-family: "Trebuchet MS", "Lucida Sans Unicode", "Lucida Grande",
    "Lucida Sans", Arial, sans-serif;
  font-size: 1.2rem;
  cursor: pointer;
  position: absolute;
  font-weight: bold;
  right: 0.5rem;
  bottom: 0.25rem;
  color: rgba(98, 96, 96, 0.68);
}

.dataList {
  background-color: #f0ebebc5;
  padding: 0rem 0.25rem;
  margin-top: -1rem;
  border: 1px solid #e7e7e7;
  font-size: 0.8rem;
}

.dataList ul {
  padding: 0;
  margin: 0;
}

.dataList > li {
  list-style-type: none;
  padding: 0.225rem;
  cursor: pointer;
}

.dataList > li:hover {
  font-weight: bold;
}
