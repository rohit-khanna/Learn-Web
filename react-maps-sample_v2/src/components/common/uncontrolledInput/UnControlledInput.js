import React, { Component } from "react";
import "./UnControlledInput.css";
import { PlacesService } from "../../../services/";
const placesEndpoint = process.env.PLACES_API_URL;

class UnControlledInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      dataList: [],
      inputValue: "",
      showDataList: false
    };
  }

  onTextChange = ev => {
    const { value } = ev.target;

    if (this.state.inputValue !== value) {
      if (value.length > 2) {
        PlacesService.searchPlaces(value, placesEndpoint)
          .then(data => {
            this.setState({
              dataList: data.map(x => x.name),
              showDataList: true
            });
          })
          .catch(err => {
            alert(err);
          });
      } else {
        this.setState({ inputValue: value });
      }
    }
  };

  onPlaceChange = ev => {
    const { name, handleChange } = this.props;
    const { textContent } = ev.target;
    this.setState({ showDataList: false, inputValue: textContent });
    handleChange(name, textContent);
  };
  render() {
    const { name, handleClearClick, label, inProgress } = this.props;

    const { inputValue, showDataList, dataList } = this.state;
    return (
      <div>
        <div className="Input_ContainerNew form-group">
          <label htmlFor={name}>{label}</label>
          <input
            type="text"
            className="form-control"
            name={name}
            value={inputValue}
            onChange={this.onTextChange}
            disabled={inProgress}
            autoComplete="off"
          />

          <div
            className="icon"
            title="clear"
            // eslint-disable-next-line react/jsx-no-bind
            onClick={() => handleClearClick({ name })}
            hidden={inProgress}
          >
            x
          </div>
        </div>
        {showDataList && (
          <ul className="dataList" onClick={this.onPlaceChange}>
            {dataList.map(x => {
              return <li key={x}>{x}</li>;
            })}{" "}
          </ul>
        )}
      </div>
    );
  }
}

export default UnControlledInput;
