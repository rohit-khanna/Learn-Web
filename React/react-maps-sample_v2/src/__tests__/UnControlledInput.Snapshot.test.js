import React from "react";
import renderer from "react-test-renderer";
import UnControlledInput from "../components/common/uncontrolledInput";

function renderUncontrolledInput(args) {
  const defaultProps = {
    name: "test",
    label: "test",
    value: "",
    handleClearClick: () => {},
    handleChange: () => {},
    inProgress: false
  };

  const props = { ...defaultProps, ...args };

  return renderer.create(<UnControlledInput {...props} />);
}

describe("<UnControlledInput/>", () => {
  it("should match the Snapshot when autocomplete list is hidden", () => {
    const tree = renderUncontrolledInput();
    expect(tree).toMatchSnapshot();
  });

  it("should match the Snapshot when autocomplete list is shown", () => {
    const tree = renderUncontrolledInput();
    const instance = tree.root.instance;
    instance.setState({ dataList: ["some place"], showDataList: true });
    expect(tree).toMatchSnapshot();
  });
});
