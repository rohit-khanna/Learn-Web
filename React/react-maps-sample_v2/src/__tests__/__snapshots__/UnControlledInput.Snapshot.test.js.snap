// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnControlledInput/> should match the Snapshot when autocomplete list is hidden 1`] = `
<div>
  <div
    className="Input_ContainerNew form-group"
  >
    <label
      htmlFor="test"
    >
      test
    </label>
    <input
      autoComplete="off"
      className="form-control"
      disabled={false}
      name="test"
      onChange={[Function]}
      type="text"
      value=""
    />
    <div
      className="icon"
      hidden={false}
      onClick={[Function]}
      title="clear"
    >
      x
    </div>
  </div>
</div>
`;

exports[`<UnControlledInput/> should match the Snapshot when autocomplete list is shown 1`] = `
<div>
  <div
    className="Input_ContainerNew form-group"
  >
    <label
      htmlFor="test"
    >
      test
    </label>
    <input
      autoComplete="off"
      className="form-control"
      disabled={false}
      name="test"
      onChange={[Function]}
      type="text"
      value=""
    />
    <div
      className="icon"
      hidden={false}
      onClick={[Function]}
      title="clear"
    >
      x
    </div>
  </div>
  <ul
    className="dataList"
    onClick={[Function]}
  >
    <li>
      some place
    </li>
     
  </ul>
</div>
`;
